zebrad/src/commands/copy_state.rs

```diff
@@ -0,0 +1,437 @@
+//! `copy-state` subcommand - copies state from one directory to another (debug only)
+//!
+//! Copying state helps Zebra developers modify and debug cached state formats.
+//!
+//! In order to test a new state format, blocks must be identical when they are:
+//! - read from the old format,
+//! - written to the new format, and
+//! - read from the new format.
+//!
+//! The "old" and "new" states can also use the same format.
+//! This tests the low-level state API's performance.
+//!
+//! ## Command Structure
+//!
+//! Copying cached state uses the following services and tasks:
+//!
+//! Tasks:
+//!  * Old to New Copy Task
+//!    * queries the source state for blocks,
+//!      copies those blocks to the target state, then
+//!      reads the copied blocks from the target state.
+//!
+//! Services:
+//!  * Source Old State Service
+//!    * fetches blocks from the best finalized chain from permanent storage,
+//!      in the old format
+//!  * Target New State Service
+//!    * writes best finalized chain blocks to permanent storage,
+//!      in the new format
+//!    * only performs essential contextual verification of blocks,
+//!      to make sure that block data hasn't been corrupted by
+//!      receiving blocks in the new format
+//!    * fetches blocks from the best finalized chain from permanent storage,
+//!      in the new format
+
+use std::{cmp::min, path::PathBuf};
+
+use abscissa_core::{config, Command, FrameworkError, Options, Runnable};
+use color_eyre::eyre::{eyre, Report};
+
+use tokio::time::Instant;
+use tower::{Service, ServiceExt};
+use zebra_chain::{block::Height, parameters::Network};
+use zebra_state as old_zs;
+use zebra_state as new_zs;
```

Comment:
🤯

Reply:
Yep, it just works, with whatever state versions we like.

If we're using two different `zebra` checkouts, we'll also need a `new_zc` and `old_zc`. And we'll need to turn blocks into bytes when copying and comparing them. But we can make those changes if we ever need to use two different state versions.

Reply:
If `zebra-chain` hasn't changed, we can just use two `zebra-state` directories based on the same `zebra-chain`. And the current `copy-state` code should work fine.